crates/vm/levm/src/vm.rs

```diff
@@ -255,6 +256,54 @@ impl VM {
             self.env.refunded_gas,
         );
 
+        if is_precompile(&current_call_frame.code_address) {
+            let precompile_result = execute_precompile(current_call_frame);
+
+            match precompile_result {
+                Ok(output) => {
+                    self.call_frames.push(current_call_frame.clone());
+
+                    return Ok(TransactionReport {
+                        result: TxResult::Success,
+                        new_state: self.cache.clone(),
+                        gas_used: current_call_frame.gas_used.low_u64(),
+                        gas_refunded: 0,
+                        output,
+                        logs: current_call_frame.logs.clone(),
+                        created_address: None,
+                    });
+                }
+                Err(error) => {
+                    if error.is_internal() {
+                        self.call_frames.push(current_call_frame.clone());
```

Comment:
If it is an internal error we don't want to push the current callframe. Just propagate it.

Reply:
Done in [Dont propagate callframe in internal error](https://github.com/lambdaclass/ethrex/pull/1454/commits/15bcde30b4a7f6ce9ebafc200377a22eae721e09)